composables/useMultipleBalance.ts

```diff
@@ -0,0 +1,140 @@
+  const refetchMultipleBalanceTimer = ref()
+  const identityStore = useIdentityStore()
+  const fiatStore = useFiatStore()
+
+  const {
+    multiBalances,
+    multiBalanceAssets,
+    multiBalanceAssetsTestnet,
+    multiBalanceNetwork,
+  } = storeToRefs(identityStore)
+
+  const currentNetwork = computed(() =>
+    isTestnet.value ? 'test-network' : 'main-network',
+  )
+  function calculateUsd(amount: string, token = 'KSM') {
+    if (!amount) {
+      return 0
+    }
+
+    const amountToNumber = Number(amount.replace(/\,/g, ''))
+
+    return calculateExactUsdFromToken(
+      amountToNumber,
+      Number(fiatStore.getCurrentTokenValue(token)),
+    )
+  }
+
+  async function getBalance(chainName: string, token = 'KSM', tokenId = 0) {
+    const currentAddress = accountId.value
+    const prefix = networkToPrefix[chainName]
+    const chain = CHAINS[prefix]
+
+    const defaultAddress = toDefaultAddress(currentAddress)
+    const publicKey = decodeAddress(currentAddress)
+    const prefixAddress = encodeAddress(publicKey, chain.ss58Format)
+    const wsProvider = new WsProvider(ENDPOINT_MAP[prefix])
+
+    const api = await ApiPromise.create({
+      provider: wsProvider,
+    })
+
+    let currentBalance
+    let nativeBalance
+
+    if (tokenId) {
+      nativeBalance = (
+        (await api.query.tokens.accounts(
+          prefixAddress,
+          tokenId,
+        )) as PalletBalancesAccountData
+      ).free.toString()
+      currentBalance = format(nativeBalance, chain.tokenDecimals, false)
+    } else {
+      nativeBalance = await balanceOf(api, prefixAddress)
+      currentBalance = format(nativeBalance, chain.tokenDecimals, false)
+    }
+
+    let selectedTokenId = String(tokenId)
+    if (chainName === 'basilisk') {
+      selectedTokenId = await getAssetIdByAccount(api, prefixAddress)
+    }
+
+    const usd = calculateUsd(currentBalance, token)
+
+    identityStore.setMultiBalances({
+      address: defaultAddress,
+      chains: {
+        [chainName]: {
+          [token.toLowerCase()]: {
+            address: prefixAddress,
+            balance: currentBalance,
+            nativeBalance,
+            usd,
+            selected: selectedTokenId === String(tokenId),
+          },
+        },
+      },
+      chainName,
+    })
+
+    identityStore.multiBalanceNetwork = currentNetwork.value
+
+    await wsProvider.disconnect()
+  }
+
+  const fetchMultipleBalance = async () => {
+    await fiatStore.fetchFiatPrice()
+    const assets = isTestnet.value
+      ? multiBalanceAssetsTestnet.value
+      : multiBalanceAssets.value
+
+    assets.forEach((item) => {
+      getBalance(item.chain, item.token, Number(item.tokenId))
+    })
+  }
+
+  onMounted(async () => {
+    if (currentNetwork.value !== multiBalanceNetwork.value) {
+      identityStore.resetMultipleBalances()
+    }
+
+    fetchMultipleBalance()
+    refetchMultipleBalanceTimer.value = setInterval(() => {
+      fetchMultipleBalance()
+    }, 30000)
```

Comment:
what's this?

Reply:
It updates the user's balance on different chains every 30 seconds.

Reply:
Won't it be resource-hungry? what happen if wss endpoint is dead/off?

Reply:
> Won't it be resource-hungry?

Without this update, the balance would not change anymore after first fetching. We introduced this regular update to solve the previous issue about incorrect balance.

>  what happen if wss endpoint is dead/off?

Then the balance from the specific endpoint, would not displayed of course.

Reply:
> Then the balance from the specific endpoint, would not displayed of course.

without blocking others?

Reply:
> > Then the balance from the specific endpoint, would not displayed of course.
> 
> without blocking others?

No. It's the same as what we do in the production environment.

Reply:
thank you for these clarifications